go/worker/compute/executor/committee/txnscheduling/algorithm/batching/incoming_queue.go

```diff
@@ -86,6 +86,28 @@ func (q *incomingQueue) addCallLocked(call []byte, callHash hash.Hash) {
 	q.queueSizeBytes += uint64(len(call))
 }
 
+func (q *incomingQueue) RemoveBatch(batch [][]byte) error {
+	q.Lock()
+	defer q.Unlock()
+
+	var newQueue [][]byte
+	for _, call := range batch {
+		callHash := hash.NewFromBytes(call)
+		if _, ok := q.callHashes[callHash]; !ok {
+			continue
+		}
+		delete(q.callHashes, callHash)
+		for _, existing := range q.queue {
```

Comment:
This seems like an expensive way to update the list as it seems like O(n^2). You could optimize it to O(2n) by just moving this loop out (since you check against `q.callHashes` anyway). Hmm, cctually it looks like this would duplicate transactions in the queue?

Reply:
yes of course (it was late when i pushed this :grin:)